packages/usa-combo-box/src/index.js

```diff
@@ -382,28 +382,63 @@ const displayList = (el) => {
   const filter = comboBoxEl.dataset.filter || DEFAULT_FILTER;
   const regex = generateDynamicRegExp(filter, inputValue, comboBoxEl.dataset);
 
-  const options = [];
-  for (let i = 0, len = selectEl.options.length; i < len; i += 1) {
-    const optionEl = selectEl.options[i];
-    const optionId = `${listOptionBaseId}${options.length}`;
-
-    if (
-      optionEl.value &&
-      (disableFiltering ||
-        isPristine ||
-        !inputValue ||
-        regex.test(optionEl.text))
-    ) {
-      if (selectEl.value && optionEl.value === selectEl.value) {
-        selectedItemId = optionId;
+  let options = [];
+  const optionsAlphabetical = [];
+  const optionsContains = [];
+  const optionList = [...selectEl.options];
+
+  /**
+   * Builds options array.
+   *
+   * If disableFiltering or a selection has been made: do not filter.
+   * Else filter options and sort by alphabetical matches followed by
+   * options that contain the query.
+   *
+   * @param {HTMLOptionElement} option - Option element from select array
+   */
+  const buildOptionsArray = (option) => {
+    if (disableFiltering || isPristine) {
+      options.push(option);
+    } else {
+      if (option.text.toLowerCase().startsWith(inputValue)) {
```

Comment:
**note:** Opted for `option.text` here instead of `option.value` because the text value is what the user sees. The option value is sometimes different than the text presented to the user. For example, components that use using abbreviations for states but the full state name on the user-facing side of the component.

Reply:
Resolving this comment because it appears to not apply anymore. Both `develop` and this branch use `option.text` for matching. (More detail in [this comment](https://github.com/uswds/uswds/pull/6122#issuecomment-2465721547))